monitor.cpp

```diff
@@ -1,38 +1,126 @@
 #include "./monitor.h"
-#include <assert.h>
+#include <iostream>
 #include <thread>
 #include <chrono>
-#include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+// Concrete implementations of vital signs
+class TemperatureVital : public VitalSign {
+public:
+    bool isInRange(float value) const override {
+        return value >= 95.0f && value <= 102.0f;
+    }
+    std::string getName() const override { return "Temperature"; }
+    std::string getAlertMessage() const override { return "Temperature is critical!"; }
+};
+
+class PulseRateVital : public VitalSign {
```

Comment:
This method of deriving one class for each vital will need many classes when we add more vitals.
Notice that `PulseRateVital` is very similar to `TemperatureVital`, except for the limits, name and message. That indicates that it can be one class with two instances of it.
Make a new class to express change in behavior. So you can have two classes, one with both high and low, one with only low?